tests/models/test_causal_self_attention.py

```diff
@@ -26,27 +34,80 @@ def _get_random_attention_layer(n_head_q, n_head_kv, n_embd, block_size, attenti
     return self_attention_layer
 
 
+@pytest.mark.parametrize(
+    "n_head_q, n_head_kv, n_embd",
+    [
+        (4, 4, 32),  # MHA (multi head attention)
+        (32, 32, 768),  # MHA (multi head attention)
+        (4, 2, 32),  # GQA (group query attention)
+        (8, 2, 32),  # GQA
+        (32, 4, 768),  # GQA
+    ],
+)
+def test_repeat_kv_heads(n_head_q, n_head_kv, n_embd):
+    batch_size = 2
+    block_size = 10
+    head_dim = n_embd // n_head_q
+    AttentionConfig(qkv_transforms=[])
+
+    q = torch.rand(batch_size, n_head_q, block_size - 1, head_dim, dtype=torch.bfloat16).cuda()
```

Comment:
As an idea: instead of torch.rand we could also do `torch.arange(0,batch_size*n_head_q*(block_size-1)*head_dim).reshape(batch_size, n_head_q, block_size - 1, head_dim)`. In this case, we can check for equality instead of approx. equality.